community/kernel/src/main/java/org/neo4j/kernel/impl/api/KernelTransactionImplementation.java

```diff
@@ -473,9 +436,22 @@ public long closeTransaction() throws TransactionFailureException
         }
     }
 
-    public boolean isClosing()
+    private void markTransactionAsClosing() throws TransactionFailureException
     {
-        return closing;
+        if ( !transactionStatus.closing() )
+        {
+            assertTransactionOpen();
+            if ( transactionStatus.isShutdown() )
+            {
+                throw new TransactionFailureException( Status.General.DatabaseUnavailable,
+                        "Transaction terminated since database is shutting down." );
+            }
+            else
+            {
+                throw new TransactionFailureException( Status.Transaction.TransactionTerminated,
+                        "Transaction is already closing. Repeated execution of transactions are not allowed." );
```

Comment:
I wonder if this message is correct. Above we've checked `transactionStatus.closing()`, which at this point must be false, so this transaction isn't in the process of closing.

Also status is set as `TransactionTerminated`, can we be sure of that?